trimesh/voxel/base.py

```diff
@@ -0,0 +1,357 @@
+            raise ValueError(
+                'encoding must be binary, got %s' % encoding.dtype)
+        self._data['encoding'] = encoding
+
+    @property
+    def _transform(self):
+        return self._data['transform']
+
+    @property
+    def transform(self):
+        """4x4 homogeneous transformation matrix."""
+        return self._transform.matrix
+
+    @transform.setter
+    def transform(self, matrix):
+        """4x4 homogeneous transformation matrix."""
+        self._transform.matrix = matrix
+
+    @property
+    def translation(self):
+        """Location of voxel at [0, 0, 0]."""
+        return self._transform.translation
+
+    @property
+    def origin(self):
+        """Deprecated. Use `self.translation`."""
+        # DEPRECATED. Use translation instead
+        return self.translation
+
+    @property
+    def scale(self):
+        """
+        3-element float representing per-axis scale.
+
+        Raises a `RuntimeError` if `self.transform` has rotation or
+        shear components.
+        """
+        return self._transform.scale
+
+    @property
+    def pitch(self):
+        """
+        Uniform scaling factor representing the side length of each voxel.
+
+        Raises a `RuntimeError` if `self.transformation` has rotation or shear
+        components of has non-uniform scaling.
+        """
+        return self._transform.pitch
+
+    @property
+    def element_volume(self):
+        return self._transform.unit_volume
+
+    def apply_transform(self, matrix):
+        self._transform.apply_transform(matrix)
+        return self
+
+    def strip(self):
+        """
+        Mutate self by stripping leading/trailing planes of zeros.
+
+        Returns
+        --------
+        self after mutation occurs in-place
+        """
+        encoding, padding = self.encoding.stripped
+        self.encoding = encoding
+        self._transform.matrix[:3, 3] = self.indices_to_points(padding[:, 0])
+        return self
+
+    @caching.cache_decorator
+    def bounds(self):
+        points = self.points
+        bounds = np.array([points.min(axis=0), points.max(axis=0)])
+        bounds.flags.writeable = False
+        return bounds
+
+    @caching.cache_decorator
+    def extents(self):
+        bounds = self.bounds
+        extents = bounds[1] - bounds[0]
+        extents.flags.writeable = False
+        return extents
+
+    @caching.cache_decorator
+    def is_empty(self):
+        return self.encoding.is_empty
+
+    @property
+    def shape(self):
+        """3-tuple of ints denoting shape of occupancy grid."""
+        return self.encoding.shape
+
+    @caching.cache_decorator
+    def filled_count(self):
+        """int, number of occupied voxels in the grid."""
+        return self.encoding.sum.item()
+
+    def is_filled(self, point):
+        """
+        Query points to see if the voxel cells they lie in are filled or not.
+
+        Parameters
+        ----------
+        point: (n, 3) float, point(s) in space
```

Comment:
In case it wasn't clear from my original documentation, the (..., 3) was meant to indicate any number of leading dimensions.

Reply:
Heh it takes arrays with `len(points.shape) == 2 and points.shape[1] == 3` right? Not arrays with shape `(a, b, c, d, e, f, g)` (etc)?

Reply:
If I haven't stuffed up it should take arrays with shape `(a, b, c, d, e, f, g, 3)` and return arrays of shape `(a, b, c, d, e, f, g)`. This makes usage like in `revoxelized` slightly more convenient.